R/postprocess.R

```diff
@@ -403,3 +382,77 @@ enw_quantiles_to_long <- function(posterior) {
   long[, quantile := as.numeric(quantile) / 100]
   return(long[])
 }
+
+#' Build the ord_obs `data.table`.
+#'
+#' @param obs Observations as pulled from nowcast$latest[[1]].
+#' @param max_delay Whole number representing the maximum delay
+#' in units of the timestep.
+#' @param internal_timestep The internal timestep in days.
+#' @param timestep The timestep to be used. This can be a string
+#' ("day", "week", "month") or a numeric whole number representing
+#' the number of days.
+#' @param sample String, "get_sample" or "no_sample".
+#' @param nowcast If getting posterior samples, the fit to pull
+#' the draws from.
+#'
+#' @return ord_obs A `data.table`.
+#'
+#' @family postprocess
+
+build_ord_obs <- function(obs, max_delay, internal_timestep, timestep, sample, nowcast = NULL) { # nolint
+  ord_obs <- coerce_dt(
+    obs, required_cols = c("reference_date", "confirm"), group = TRUE
+  )
+  check_timestep_by_group(
+    ord_obs, "reference_date", timestep, exact = TRUE
+  )
+
+  ord_obs <- subset_obs(ord_obs, max_delay, internal_timestep,
+                        select = "after")
+
+sample <- rlang::arg_match(sample, c("get_sample", "no_sample"))
+
+  data.table::setorderv(ord_obs, c(".group", "reference_date"))
+  if (sample == "get_sample") {
+    ord_obs <- data.table::data.table(
+      .draws = 1:max(nowcast$.draw),
+      obs = rep(list(ord_obs), max(nowcast$.draw))
+    )
+    ord_obs <- ord_obs[, rbindlist(obs), by = .draws]
+    ord_obs <- ord_obs[order(.group, reference_date)]
+  }
+  return(ord_obs)
+}
+
+#' Subset observations data table for either modelled dates
+#' or not-modelled earlier dates.
+#'
+#' @param ord_obs The observations `data.table` to be subset,
+#' as pulled from the result of calling epinowcast() and
```

Comment:
```suggestion
#' as pulled from the result of calling [epinowcast()] and
```

(makes it a link in the rendered docs)